### ISSUE TITLE
Upgrade to v2.15.19 breaks spatie settings plugin with casted attributes

### ISSUE THREAD

**Issue by bernhardh** [filamentphp/filament#3570]
### Package

filament/spatie-laravel-settings-plugin

### Package Version

v2.15.19

### Laravel Version

v9.25.1

### Livewire Version

v2.10.7

### PHP Version

8.1.9

### Problem description

I am using the spatie-settings plugin with casted attributes, just as described in the spatie docs and it works outside the admin without any problems and it was working in filament till incl. `v2.15.18`. The issue is because of the commit #3495

It doesn't show the form value.

### Expected behavior

The settings plugin should show the data stored in the settings table correctly. 

I have created a demo repository to reproduce it and here is a video how it worked in `v2.15.18` and how it looks like after upgrade:

https://user-images.githubusercontent.com/642292/185605315-ecf5df0c-7f45-4191-905a-dd556a867c52.mp4

The problem here is, that the `NavigationItemsCast` class returns an array of `NavigationItem`. If I change:
```
$items[] = NavigationItem::make($row);
```
to
```
$items[] = NavigationItem::make($row)->toArray();
```
It works again, but then I have array's instead of objects (and their methods) in the frontend as well.

### Steps to reproduce

1. Install the demo repository (its a fork of you admin demo) and use the `issue/spatie-settings-not-working` branch and run migrations (example data for the settings is included).
2. Go to `Manage General Setting` page. 

# Expected behavior (just like in  `v2.15.18`)

![image](https://user-images.githubusercontent.com/642292/185606815-7a7dd18c-1d89-4e28-bc8a-cfd9a6e24ae9.png)

# Current behavior
![image](https://user-images.githubusercontent.com/642292/185606669-b796f64e-656f-4f86-906c-915afb9b873b.png)

# Sidenote

When you add new values or fill up the empty input fields and click save afterwards, the values are show as expected. But after page refresh, they are gone again.

### Reproduction repository

https://github.com/bernhardh/filamentdemo

**Comment by ekandreas** [comment]
Great! I can also confirm this issue not being trusted in previous posting.

**Comment by danharrin** [comment]
Fixed by 8453d71d9c3f06a1c403dff706b7b3a563484fa0.
